### PR TITLE
Fix several spelling corrections + add refinement

### PR DESCRIPTION
**Pull request by luzpaz**

### CODE REVIEW

File: codespell_lib/data/dictionary.txt [hunk 1]
```diff
@@ -4123,6 +4123,7 @@ bootstap->bootstrap
 bootstapped->bootstrapped
 bootstapping->bootstrapping
 bootstaps->bootstraps
+booundary->boundary
```
Comment (vikivivi) [inline codespell_lib/data/dictionary.txt:4126]: Mind adding this?
https://github.com/search?q=booundaries&type=code
```text
booundaries->boundaries
```

**Review by vikivivi — CHANGES_REQUESTED** [review]

File: codespell_lib/data/dictionary.txt [hunk 2]
```diff
@@ -20922,6 +20923,8 @@ parametes->parameters
 parametic->parametric, paramedic,
 parametised->parametrised
 parametr->parameter
+parametre->parameter
+parametres->parameters
```
Comment (vikivivi) [inline codespell_lib/data/dictionary.txt:20927]: Also https://github.com/search?q=parametreless&type=code
```text
parametreless->parameterless
```

**Review by vikivivi — CHANGES_REQUESTED** [review]

File: codespell_lib/data/dictionary.txt (hunk 2 above)
Comment (luzpaz) [reply codespell_lib/data/dictionary.txt:20931]: 30b3c38

File: codespell_lib/data/dictionary.txt (hunk 1 above)
Comment (luzpaz) [reply codespell_lib/data/dictionary.txt:4127]: 30b3c38

**Pull request merged**